hardware/EnphaseAPI.cpp

```diff
@@ -227,6 +227,55 @@ int EnphaseAPI::getSunRiseSunSetMinutes(const bool bGetSunRise)
 	return 0;
 }
 
+// emupwGetPasswdForSn / emupwGetMobilePasswd taken from https://github.com/sarnau/EnphaseEnergy
+static std::string emupwGetPasswdForSn(const std::string &serialNumber, const std::string &userName, const std::string &realm)
+{
+    std::string szMD5 = GenerateMD5Hash(std::string("[e]") + userName + "@" + realm + "#" + serialNumber + " EnPhAsE eNeRgY ");
+
+	return szMD5;
+}
+
+static std::string emupwGetMobilePasswd(const std::string &serialNumber, const std::string &userName, const std::string &realm)
```

Comment:
Please make this a member of the class and not static
Also maybe rename this to V5_GetMobilePassword

Reply:
done, I leave the emupw prefix to stick with the original function name from the library.